src/OpenLoco/src/Windows/ObjectSelectionWindow.cpp

```diff
@@ -99,21 +159,112 @@ namespace OpenLoco::Ui::Windows::ObjectSelectionWindow
 
     static WindowEventList _events;
 
-    // 0x00473154
-    static void sub_473154(Window* self)
+    // 0x0047322A
+    static void rotateTabs(uint8_t newStartPosition)
     {
-        registers regs;
-        regs.esi = X86Pointer(self);
-        call(0x00473154, regs);
+        TabPosition newPosInfo[36] = {};
+        uint8_t j = 0;
+
+        for (uint8_t i = newStartPosition; _tabInformation[i].index != 0xFF; i++)
+        {
+            newPosInfo[j].index = _tabInformation[i].index;
+            newPosInfo[j].row = j < 19 ? 0 : 1;
+            j++;
+        }
+
+        for (uint8_t i = 0; i < newStartPosition; i++)
+        {
+            newPosInfo[j].index = _tabInformation[i].index;
+            newPosInfo[j].row = j < 19 ? 0 : 1;
+            j++;
+        }
+
+        // Add a marker to denote the last tab
+        newPosInfo[j].index = 0xFF;
+
+        // Copy new tab order into _tabInformation
+        std::memcpy(_tabInformation, newPosInfo, std::size(newPosInfo) * sizeof(TabPosition));
     }
 
     // 0x004731EE
-    static void sub_4731EE(Window* self, ObjectType eax)
+    static void repositionTargetTab(Window* self, ObjectType targetTab)
     {
-        registers regs;
-        regs.eax = static_cast<uint32_t>(eax);
-        regs.esi = X86Pointer(self);
-        call(0x004731EE, regs);
+        self->currentTab = enumValue(targetTab);
+        for (auto i = 0U; i < std::size(_tabInformation); i++)
+        {
+            // Ended up in a position without info? Reassign positions first.
+            if (_tabInformation[i].index == 0xFF)
+            {
+                self->var_856 |= (1 << 0);
+                assignTabPositions(self);
+                continue;
+            }
+
+            if (_tabInformation[i].index == enumValue(targetTab))
+            {
+                // Found current tab, and its in bottom row? No change required
+                if (_tabInformation[i].row == 0)
+                    return;
+                // Otherwise, we'll rotate the tabs around, such that this one is in the bottom row
+                else
+                    return rotateTabs(i);
+            }
+        }
+    }
+
+    // 0x00473154
+    static void assignTabPositions(Window* self)
+    {
+        uint8_t currentRow = 0;
+        uint8_t currentPos = 0;
+        uint8_t rowCapacity = 19;
```

Comment:
This constant is used in quite a few locations. Can you yank it out into a constexpr.